src/managers/SessionLockManager.cpp

```diff
@@ -126,19 +125,28 @@ void CSessionLockManager::onNewSessionLock(wlr_session_lock_v1* pWlrLock) {
         pWlrLock, "wlr_session_lock_v1");
 
     wlr_session_lock_v1_send_locked(pWlrLock);
+
+    g_pSessionLockManager->activateLock();
 }
 
 bool CSessionLockManager::isSessionLocked() {
     return m_sSessionLock.active;
 }
 
+bool CSessionLockManager::shouldLockSurfaceBeMapped(SSessionLockSurface* sls) {
+    if (!sls)
+        return false;
+
+    if (sls->timeAlive.getSeconds() > 1.f)
+        return true;
+
+    return false;
+}
+
 SSessionLockSurface* CSessionLockManager::getSessionLockSurfaceForMonitor(const int& id) {
     for (auto& sls : m_sSessionLock.vSessionLockSurfaces) {
         if (sls->iMonitorID == id) {
-            if (sls->mapped)
```

Comment:
I would not make this change, functions calling this expect the surf to be mapped

Instead of this you can make `shouldLockSurfaceBeMapped` take a cmonitor*

Reply:
I am just going to use the monitor id instead. Are you implicitly casting the monitor id from `uint64_t` to `int` on purpose btw? (via `const int&`)